src/Control/HTTPRequest.php

```diff
@@ -810,7 +812,8 @@ public function getIP()
      */
     public function setIP($ip)
     {
-        if (!filter_var($ip, FILTER_VALIDATE_IP)) {
+        // We can't use ConstraintValidator here because it relies on injector and the kernel may not have booted yet.
+        if (!IpValidator::checkIp($ip, Ip::ALL)) {
```

Comment:
This ends up being the exact same thing under the hood - though frankly Symfony have a better track record of keeping up with RFCs than PHP internal functions do so this seems better long-term.